### ISSUE TITLE
Multiple occurrences in open_constr

### ISSUE THREAD

**Issue by eponier** [coq/coq#9320]
#### Version

Coq 8.7.2 and development version

#### Operating system

Fedora

#### Description of the problem

Due to [this question](https://stackoverflow.com/questions/54081136/match-context-pattern-inside-a-tactic-tactic-notation), I considered using `open_constr` to create a matching pattern.

```
Goal 2 = 3 -> False.
Proof.
  let k := open_constr:(?[n] = ?n) in
  match goal with | |- context [ k ] => idtac end.
```

This Ltac code works, while `n` cannot be instantiated both to `2` and `3`. I find this weird.

Is this a strange behaviour, or a misuse of `open_constr` or something else I don't understand?

**Comment by JasonGross** [comment]
I believe evars in open_constr / holes in uconstr become underscores in match patterns which match anything.  Perhaps user-named evars should become binding patterns?

cc @herbelin

**Comment by ppedrot** [comment]
Yep, that's due to the dreaded `pattern_of_constr` function, which is responsible for one the nastiest source of evar-sensitivity in the Coq codebase.
